aws/resource_aws_kinesis_analyticsv2_application.go

```diff
@@ -0,0 +1,2177 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesisanalyticsv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+var (
+	validateKinesisAnalyticsV2Runtime = validation.StringInSlice([]string{
+		kinesisanalyticsv2.RuntimeEnvironmentSql10,
+		kinesisanalyticsv2.RuntimeEnvironmentFlink16,
+		kinesisanalyticsv2.RuntimeEnvironmentFlink18,
+	}, false)
+
+	validateKinesisAnalyticsV2MetricsLevel = validation.StringInSlice([]string{
+		kinesisanalyticsv2.MetricsLevelApplication,
+		kinesisanalyticsv2.MetricsLevelOperator,
+		kinesisanalyticsv2.MetricsLevelParallelism,
+		kinesisanalyticsv2.MetricsLevelTask,
+	}, false)
+
+	validateKinesisAnalyticsV2LogLevel = validation.StringInSlice([]string{
+		kinesisanalyticsv2.LogLevelInfo,
+		kinesisanalyticsv2.LogLevelWarn,
+		kinesisanalyticsv2.LogLevelError,
+		kinesisanalyticsv2.LogLevelDebug,
+	}, false)
+
+	validateKinesisAnalyticsV2ConfigurationType = validation.StringInSlice([]string{
+		kinesisanalyticsv2.ConfigurationTypeCustom,
+		kinesisanalyticsv2.ConfigurationTypeDefault,
+	}, false)
+)
+
+func resourceAwsKinesisAnalyticsV2Application() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsKinesisAnalyticsV2ApplicationCreate,
+		Read:   resourceAwsKinesisAnalyticsV2ApplicationRead,
+		Update: resourceAwsKinesisAnalyticsV2ApplicationUpdate,
+		Delete: resourceAwsKinesisAnalyticsV2ApplicationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				arns := strings.Split(d.Id(), ":")
+				name := strings.Replace(arns[len(arns)-1], "application/", "", 1)
+				d.Set("name", name)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_execution_role": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"runtime": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateKinesisAnalyticsV2Runtime,
+			},
+
+			"s3_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"s3_object": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"code_content_type": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Add `ValidateFunc` to check for `kinesisanalyticsv2.CodeContentTypePlaintext` or `kinesisanalyticsv2.CodeContentTypeZipfile`.